src/apb/engine.py

```diff
@@ -464,84 +465,146 @@ def relist_service_broker(kwargs):
+                    'name': name,
+                    'namespace': namespace,
+                },
+                'subjects': [{
+                    'kind': 'ServiceAccount',
+                    'name': service_account,
+                    'namespace': namespace,
                 }],
-                'restartPolicy': 'Never',
-                'serviceAccountName': 'service-account-1',
+                'roleRef': {
+                    'name': role,
+                },
             }
+        )
+    except ApiException as e:
+        raise e
+    except Exception as e:
+        # TODO:
+        # Right now you'll see something like --
+        #   Exception occurred! 'module' object has no attribute 'V1RoleBinding'
+        # Looks like an issue with the openshift-restclient...well the version
+        # of k8s included by openshift-restclient
+        pass
+    print("Created Role Binding")
+    return name
 
-        }
-        create_service_account()
-        create_role_binding()
-        api.create_namespaced_pod("default", pod_manifest)
+
+def create_pod(image, name, namespace, command, service_account):
+    print("Creating pod with image {} in {}".format(image, namespace))
+    try:
+        kubernetes_config.load_kube_config()
+        api = kubernetes_client.CoreV1Api()
+        pod = api.create_namespaced_pod(
+            namespace,
+            {
+                'apiVersion': 'v1',
+                'kind': 'Pod',
+                'metadata': {
+                    'generateName': name,
+                    'namespace': namespace
+                },
+                'spec': {
+                    'containers': [{
+                        'image': image,
+                        'imagePullPolicy': 'IfNotPresent',
+                        'name': name,
+                        'command': command,
+                        'env': [
+                            {'name': 'POD_NAME',
+                             'valueFrom': {'fieldRef': {'fieldPath': 'metadata.name'}}
+                            },
+                            {'name': 'POD_NAMESPACE',
+                             'valueFrom': {'fieldRef': {'fieldPath': 'metadata.namespace'}}
+                            }
+                        ],
+                    }],
+                    'restartPolicy': 'Never',
+                    'serviceAccountName': service_account,
+                }
+            }
+        )
         print("Created Pod")
+        return pod.metadata.name
     except Exception as e:
         print("failed - %s" % e)
 
 
+def run_apb(project, image, name, action, parameters={}):
+    ns = create_project(project)
+    sa = create_service_account(ns)
+    rb = create_role_binding(ns, sa)
+
+    parameters['namespace'] = ns
+    command = ['entrypoint.sh', action, "--extra-vars", json.dumps(parameters)]
+
+    po = create_pod(
+        image=image,
+        name=name,
+        namespace=ns,
+        command=command,
+        service_account=sa
+    )
+
```

Comment:
This might fail flake8 if you don't have 2 newlines.

Reply:
Fixed.